src/producer.rs

```diff
@@ -2062,3 +2107,125 @@ impl<B: BufRead> Iterator for BinaryLines<B> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use csv::{ReaderBuilder, StringRecord};
+    use giganto_client::ingest::network::Conn;
+
+    use super::*;
+
+    fn create_zeek_record(timestamp: &str, uid: &str) -> StringRecord {
+        let data = format!("{timestamp}\t{uid}\t192.168.1.77\t57655\t209.197.168.151\t1024\ttcp\tirc-dcc-data\t2.256935\t124\t42208\tSF\t-\t-\t0\tShAdDaFf\t28\t1592\t43\t44452\t-");
+        ReaderBuilder::new()
+            .delimiter(b'\t')
+            .has_headers(false)
+            .from_reader(data.as_bytes())
+            .into_records()
+            .next()
+            .unwrap()
+            .unwrap()
+    }
+
+    #[test]
+    fn test_zeek_timestamp_deduplication() {
+        // Test that duplicate timestamps get incremented by 1 nanosecond
+        let timestamp = "1562093121.655728"; // Same timestamp for both records
+
+        let record1 = create_zeek_record(timestamp, "C6nQDk3TAW2xcVQtQ1");
+        let record2 = create_zeek_record(timestamp, "C6nQDk3TAW2xcVQtQ2");
+
+        // Parse both records
+        let (_, ts1) = Conn::try_from_zeek_record(&record1).unwrap();
+        let (_, ts2) = Conn::try_from_zeek_record(&record2).unwrap();
+
+        // Both should have the same timestamp initially (before deduplication)
+        assert_eq!(ts1, ts2);
+
+        // Test the deduplication logic manually
+        let mut reference_timestamp: Option<i64> = None;
+        let mut timestamp_offset = 0_i64;
+
+        // Process first record
+        if let Some(ref_ts) = reference_timestamp {
+            if ts1 == ref_ts {
+                timestamp_offset += 1;
+            } else {
+                reference_timestamp = Some(ts1);
+                timestamp_offset = 0;
+            }
+        } else {
+            reference_timestamp = Some(ts1);
+        }
+        let final_ts1 = ts1 + timestamp_offset;
+
+        // Process second record (same timestamp)
+        if let Some(ref_ts) = reference_timestamp {
+            if ts2 == ref_ts {
+                timestamp_offset += 1;
+            } else {
+                timestamp_offset = 0;
+            }
+        }
+        let final_ts2 = ts2 + timestamp_offset;
+
+        // After deduplication, second timestamp should be incremented by 1 nanosecond
+        assert_eq!(final_ts2, final_ts1 + 1);
+    }
+
+    #[test]
+    fn test_zeek_different_timestamps_no_deduplication() {
```

Comment:
The same Timestamp dedup logic is implemented in the `test_zeek_timestamp_deduplication()` function and the `test_zeek_different_timestamps_no_deduplication()` function. Please implement this part separately as a function.